resources/public/api/conf/2.0/application.raml

```diff
@@ -0,0 +1,64 @@
+#%RAML 2.0
+---
+title: API Platform Test
+version: 2.0
+
+protocols: [ HTTPS ]
+baseUri: https://api.service.hmrc.gov.uk/
+
+documentation:
+ - title: Overview
+   content: !include docs/overview.md
+
+mediaType: application/json
+
+uses:
+  sec: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/securitySchemes.raml
+  headers: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/headers.raml
+  annotations: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/annotations.raml
+  types: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/types.raml
+
+/api-platform-test:
+  /:
+    get:
+      displayName: dummy resource
+      is: [headers.acceptHeader]
+
+  /ciao:
+    get:
+      displayName: dummy Ciao resource
+      is: [headers.acceptHeader]
+      queryParameters:
+        surname:
+          type: string
+          required: false
+        firstName:
+          type: string
+          required: false
+        middleName:
+          type: string
+          required: false
+
+  /ciao/{surname}:
+    uriParameters:
+      surname:
+        type: string
+
+    get:
+      displayName: dummy Ciao Surname resource
+      is: [headers.acceptHeader]
+
+  /ciao/{surname}/{firstName}:
```

Comment:
Following on from above, this should be just `/{firstName}` and only `firstName` needs to be defined in `uriParameters` since it then inherits the definition of `surname` from the level above in the hierarchy.

Reply:
Good catch.
I will make the code changes.